src/main/java/dota/buff/service/impl/HeroServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package dota.buff.service.impl;
+
+import com.github.wannesvr.core.model.hero.Hero;
+
+import dota.buff.model.HeroDTO;
+import dota.buff.service.ConvertService;
+import dota.buff.service.HeroService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class HeroServiceImpl implements HeroService {
+
+    private final List<HeroDTO> heroList;
+
+    @Autowired
+    public HeroServiceImpl(ConvertService convertService, List<Hero> allHeroes) {
+        this.heroList = convertService.convertHeroList(allHeroes);
```

Comment:
I guess would be better to check here, when spring is creating an instance of service and **heroList** is null or empty -throw an exception, it will be **fail-fast behavior**, even before will be called some method from the current service.

Reply:
Useless, we can checked it in SpringConfig